yak/CMakeLists.txt

```diff
@@ -73,6 +73,11 @@ set_directory_properties( PROPERTIES COMPILE_DEFINITIONS "" )
 link_directories(${PCL_LIBRARY_DIRS})
 add_definitions(${PCL_DEFINITIONS})
 
+if(${CMAKE_VERSION} VERSION_LESS "3.9.0")
+  message(WARNING "You are using CMake version ${CMAKE_VERSION}. Consider upgrading!")
+  include_directories(include)  # TODO: Avoid using include_directories at all!
+endif()
```

Comment:
Why is this required?

Reply:
When I build using CMake 3.5 it isn't able to find the header files in the include directory. I think there might be a problem with the generator expressions in the `target_include_directories` function.

Reply:
This is the specific error:

```
Errors     << yak:make /home/joe/workspaces/yak_ws/logs/yak/build.make.003.log                                                                                                                                                        
/home/joe/workspaces/yak_ws/src/yak/yak/src/cuda/proj_icp.cu:1:39: fatal error: yak/kfusion/cuda/device.hpp: No such file or directory
compilation terminated.
/home/joe/workspaces/yak_ws/src/yak/yak/src/cuda/imgproc.cu:1:39: fatal error: yak/kfusion/cuda/device.hpp: No such file or directory
compilation terminated.
CMake Error at yak_generated_imgproc.cu.o.cmake:207 (message):
  Error generating
  /home/joe/workspaces/yak_ws/build/yak/CMakeFiles/yak.dir/src/cuda/./yak_generated_imgproc.cu.o


CMake Error at yak_generated_proj_icp.cu.o.cmake:207 (message):
  Error generating
  /home/joe/workspaces/yak_ws/build/yak/CMakeFiles/yak.dir/src/cuda/./yak_generated_proj_icp.cu.o


make[2]: *** [CMakeFiles/yak.dir/src/cuda/yak_generated_imgproc.cu.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[2]: *** [CMakeFiles/yak.dir/src/cuda/yak_generated_proj_icp.cu.o] Error 1
/home/joe/workspaces/yak_ws/src/yak/yak/src/cuda/tsdf_volume.cu:1:39: fatal error: yak/kfusion/cuda/device.hpp: No such file or directory
compilation terminated.
CMake Error at yak_generated_tsdf_volume.cu.o.cmake:207 (message):
  Error generating
  /home/joe/workspaces/yak_ws/build/yak/CMakeFiles/yak.dir/src/cuda/./yak_generated_tsdf_volume.cu.o


make[2]: *** [CMakeFiles/yak.dir/src/cuda/yak_generated_tsdf_volume.cu.o] Error 1
make[1]: *** [CMakeFiles/yak.dir/all] Error 2
make: *** [all] Error 2
cd /home/joe/workspaces/yak_ws/build/yak; catkin build --get-env yak | catkin env -si  /usr/bin/make --jobserver-fds=6,7 -j; cd -
```

Reply:
I think I found the problem: the older-style CUDA support provided in the CMake 3.5 FindCUDA module doesn't support the `target_include_directories` directive ([similar issue here](https://gitlab.kitware.com/cmake/cmake/issues/18604), [brief summary here](https://codeyarns.com/2015/03/26/target_include_directories-does-not-work-with-cuda/), [original bug report here](https://cmake.org/Bug/view.php?id=14201)).

This problem has been fixed with the improvements to CUDA support added in newer versions of CMake (3.9 and later, I think). [This PR](https://github.com/ros-industrial/yak/pull/3) uses CMake 3.10 and the "modern CMake" approach, and `target_include_directories` works just fine there.

Little bit frustrating from the perspective of working towards a cross-compatible project...